LevelUp.Pos.ProposedOrders/ProposedOrderCalculator.cs

```diff
@@ -88,23 +94,45 @@ int customerPaymentAmount
             int appliedDiscountAmount
         )
         {
-            AdjustedCheckValues values = CalculateCreateProposedOrderValues(
-                totalOutstandingAmount,
+            int totalOutstandingAmountWithDiscount = totalOutstandingAmount + Math.Abs(appliedDiscountAmount);
+
+            int adjustedSpendAmount = CalculateAdjustedCustomerPaymentAmount(
+                customerPaymentAmount, 
+                totalOutstandingAmountWithDiscount);
+
+            int adjustedTaxAmount = CalculateAdjustedTaxAmount(
+                totalOutstandingAmountWithDiscount, 
                 totalTaxAmount,
-                totalExemptionAmount,
-                customerPaymentAmount);
+                adjustedSpendAmount);
 
-            values.SpendAmount = CalculateAdjustedSpendAmountCompleteOrder(totalOutstandingAmount, 
```

Comment:
When I initially wrote this library, this felt wrong, but I couldn't put my finger on it.  It now makes sense that if we're going to adjust the `spendAmount` especially for `complete_order`, then we'll want to re-use that newly adjusted `spendAmount` to get an `adjustedTaxAmount`.  This has been fixed now.